src/ngx_pagespeed.cc

```diff
@@ -544,7 +544,8 @@ ngx_int_t copy_response_headers_to_ngx(
     }
 
     if (STR_EQ_LITERAL(name, "Cache-Control")) {
-      ps_set_cache_control(r, const_cast<char*>(value_gs.c_str()));
+      ps_set_cache_control(r, string_piece_to_pool_string(
+          r->pool, value_gs.c_str()));
```

Comment:
I think you can just use `value_s` here; it looks like that's already available here for exactly this lifetime issue?

It also might be clearer to change the code above so that `value` is the output of `string_piece_to_pool_string` and then we can do away with `value_s`.